.github/workflows/github-cxx-qt-tests.yml

```diff
@@ -88,6 +88,55 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           verbose: true
 
+  build-wasm:
+    needs: [clang_format, license_check, rust_format_check, markdown_lint, shellcheck]
+    runs-on: ubuntu-22.04
+    env:
+      SCCACHE_CACHE_SIZE: 600M
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+      - name: Setup toolchain
+        run: |
+          rustup component add clippy rustfmt
+          rustup target add wasm32-unknown-emscripten
+          git clone https://github.com/emscripten-core/emsdk
+          ./emsdk/emsdk install 2.0.14
+          ./emsdk/emsdk activate 2.0.14
+          source ./emsdk/emsdk_env.sh
+      - name: "Build Rust tools"
+        run: |
+          cargo install --no-default-features sccache
+          cargo install mdbook mdbook-linkcheck
+      - name: "Install Dependencies"
```

Comment:
add in the compiler cache as we are setting up sccache

```
    # We want our compiler cache to always update to the newest state.
    # The best way for us to achieve this is to **always** update the cache after every landed commit.
    # That way it will closely follow our development.
    # And if a PR diverges a lot with its cache that's not a big deal, as it will be merged eventually.
    #
    # This is a workaround for the fact that GH doesn't support updating existing caches.
    # See: https://github.com/azu/github-actions-overwrite-cache-example
    #
    # Ideally we'd like to use this:
    # - name: "Compiler cache"
    #   uses: actions/cache@v4
    #   with:
    #     update: true <------- THIS DOESN'T EXIST YET
    #     path: /home/runner/.cache/sccache
    #     key: "Ubuntu 22.04 (wasm_32) Qt6_compiler_cache"
    - name: "Restore Compiler Cache"
      id: compiler-cache-restore
      uses: actions/cache/restore@v4
      with:
        path: /home/runner/.cache/sccache
        key: "Ubuntu 22.04 (wasm_32) Qt6_compiler_cache"
```